lib/api/core/validation/index.js

```diff
@@ -706,27 +698,25 @@ function checkAllowedProperties(object, allowedProperties) {
  * @throws InternalError
  */
 function curateStructuredFields(typeAllowsChildren, fields, maxDepth) {
-  var
-    i,
+  const
+    /** @type StructuredFieldSpecification */
     structuredFields = {
       children: {},
       root: true
     };
 
-  for (i = 1; i <= maxDepth; i++) {
+  for (let i = 1; i <= maxDepth; i++) {
     if (!fields.hasOwnProperty(i)) {
-      throw new InternalError('All levels of an object have to be defined in the specification.');
+      throw new KuzzleInternalError('All levels of an object have to be defined in the specification.');
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 104 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)